verifier/src/main/kotlin/tech/figure/classification/asset/verifier/util/eventstream/providers/BlockApiEventStreamProvider.kt

```diff
@@ -46,14 +46,26 @@ class BlockApiEventStreamProvider(
             while (coroutineScope.isActive) {
                 (from..current).forEach { blockHeight ->
                     if (from > current) return@forEach
-                    process(blockHeight, onBlock, onEvent, onError, onCompletion)
+                    process(blockHeight, onBlock, onEvent, onError)
                     lastProcessed.set(blockHeight)
                 }
 
                 // Once we've met the current block, no need to keep spinning. Wait here for 4 seconds and process again.
                 delay(DEFAULT_BLOCK_DELAY_MS.milliseconds)
                 from = lastProcessed.incrementAndGet()
-                current = currentHeight()
+
+                runCatching {
```

Comment:
It might be worthwhile to make this retryable current height function a private function in this class.  You also fetch the current height on line 40, and it would probably be good to do some retries there as well

Reply:
:roger-roger: done and done.